File: test/org/cerion/symcalc/expression/function/integer/FactorialTest.java
package org.cerion.symcalc.expression.function.integer;

import org.cerion.symcalc.expression.number.IntegerNum;
import org.junit.Test;

import static org.junit.Assert.*;

public class FactorialTest {

    @Test
    public void basic() {
        assertEquals(new IntegerNum(1), new Factorial(new IntegerNum(0)).eval());
        assertEquals(new IntegerNum(1), new Factorial(new IntegerNum(1)).eval());
        assertEquals(new IntegerNum(2), new Factorial(new IntegerNum(2)).eval());
        assertEquals(new IntegerNum(720), new Factorial(new IntegerNum(6)).eval());
    }
}

File: symcalc/src/test/java/function/integer/FactorialTest.kt
package org.cerion.symcalc.function.integer

import org.cerion.symcalc.number.Integer
import kotlin.test.Test
import kotlin.test.assertEquals

class FactorialTest {

    @Test
    fun basic() {
        assertEquals(Integer(1), Factorial(Integer(0)).eval())
        assertEquals(Integer(1), Factorial(Integer(1)).eval())
        assertEquals(Integer(2), Factorial(Integer(2)).eval())
        assertEquals(Integer(720), Factorial(Integer(6)).eval())
    }
}